pkg/client/client.go

```diff
@@ -166,11 +175,31 @@ func (o *OptimizelyClient) GetEnabledFeatures(userContext entities.UserContext)
 // GetFeatureVariableBoolean returns the feature variable value of type bool associated with the given feature and variable keys.
 func (o *OptimizelyClient) GetFeatureVariableBoolean(featureKey, variableKey string, userContext entities.UserContext) (value bool, err error) {
 
-	val, valueType, err := o.GetFeatureVariable(featureKey, variableKey, userContext)
+	var convertedValue bool
+	val, valueType, featureDecision, err := o.getFeatureVariable(featureKey, variableKey, userContext)
+	defer func() {
```

Comment:
Nit: I would find this easier to read without a `defer`. It's referencing values that are assigned below, and above, so I have to read both directions, whereas if it was just moved down and didn't use `defer`, I only have to read one direction.

Reply:
defer makes it easier to make fully backward compatible with the previous version sent from composite_service. We have 2 returns before we get to the final one. if we remove defer then we need to send notification for each return.